react-carousel-component/my-app/src/Carousel.js

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from "react";
+import { FaAngleLeft, FaAngleRight, FaCircle, FaRegCircle, FaRegWindowClose } from "react-icons/fa";
+import './Carousel.css';
+
+export default function Carousel({ images }) {
+  const [currentImageIndex, setCurrentImageIndex] = useState(0);
+
+  const { name: currentName, imageURL: currentImage } = images[currentImageIndex];
+
+  function handleClick(index) {
+    setCurrentImageIndex(index)
+  }
+
+  function handlePrev() {
+    setCurrentImageIndex(((currentImageIndex - 1) + images.length) % images.length)
+  }
+
+  function handleNext() {
+    setCurrentImageIndex((currentImageIndex + 1) % images.length)
+  }
+
+  useEffect(() => {
+    const timeoutID = setTimeout(handleNext, 3000);
+    return () => clearInterval(timeoutID)
+  })
```

Comment:
There's no dependency array here, even though it depends on `handleNext`. That means this will execute on every render. That avoids the closure problem, I guess, but it's not usually the best way to write Effects (it will reset your counter on every render).